### ISSUE TITLE
[Research] Add E-ASR Single-Channel Artifact Rejection (preserves data instead of discarding)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#30]
## What

Implement **E-ASR (Embedded Artifact Subspace Reconstruction)** from IIT Guwahati (Sensors, Oct 2024) — adapts ASR (traditionally requiring multi-channel EEG) to work on **single channels** via delay-vector embedding. Cleans artifacts in-place without discarding entire epochs.

## Why

- Current artifact handling: simple 75 uV amplitude threshold → rejects entire epoch → **loses data**
- E-ASR: cleans artifacts in-place per channel → **preserves usable data**
- Achieves RRMSE of 43.87% and correlation coefficient of 0.91 on semi-simulated data
- Particularly important for AF7/AF8 where jaw clench/blink artifacts are most common
- Open-source: https://github.com/NeuralLabIITGuwahati/E-ASR
- No model training required — algorithmic approach

### Step 1: Integrate E-ASR into preprocessing pipeline
```python
# ml/processing/eeg_processor.py — add after bandpass, before feature extraction

from e_asr import EmbeddedASR  # or inline implementation

def clean_artifacts_easr(signal, fs=256, embedding_dim=10, threshold=5.0):
    """Clean artifacts per-channel using E-ASR.
    
    Algorithm:
    1. Create pseudo-multichannel via time-delay embedding
    2. Apply standard ASR on pseudo-multichannel matrix
    3. Reconstruct cleaned single-channel signal
    
    Args:
        signal: single-channel EEG (1D array)
        fs: sampling rate
        embedding_dim: delay embedding dimension (default 10)
        threshold: ASR rejection threshold in std devs
    
    Returns:
        cleaned_signal: artifact-corrected signal (same shape)
    """
```

### Step 2: Insert into preprocessing chain
```
Current:  raw → bandpass → notch → [75uV threshold reject] → features
Proposed: raw → bandpass → notch → [E-ASR clean per channel] → features
```

### Step 3: Replace epoch rejection with cleaning
- Instead of `skip_epoch()` on amplitude > 75 uV, apply E-ASR first
- Only reject if signal is still bad after cleaning (extreme artifact)
- Track `artifact_cleaned_ratio` in response for data quality monitoring

### Files to Modify
- `ml/processing/eeg_processor.py` — Add `clean_artifacts_easr()`, update `preprocess()`
- `ml/processing/e_asr.py` (NEW) — E-ASR implementation (or pip install)
- `ml/api/routes.py` — Add `artifact_cleaned_ratio` to `/analyze-eeg` response

### Expected Impact
- Data retention: **+30-50%** more usable epochs (currently discarded epochs recovered)
- Indirect accuracy improvement: **+5-10%** (more training data, less noisy features)
- Better real-time experience: fewer "no data" gaps during live EEG

## Complexity
LOW-MEDIUM — algorithmic, no training needed, open-source code available

## References
- Paper: https://www.mdpi.com/1424-8220/24/20/6734
- Code: https://github.com/NeuralLabIITGuwahati/E-ASR
- RRMSE: 43.87%, correlation: 0.91 on semi-simulated data
